### PR TITLE
CSHARP-2629: Remove workaround for SERVER-40685

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
evergreen: https://evergreen.mongodb.com/version/5dbaf2519ccd4e78c5aaa21f

### CODE REVIEW

File: tests/MongoDB.Driver.TestHelpers/DriverTestConfiguration.cs [hunk 1]
```diff
@@ -113,9 +111,15 @@ public static DisposableMongoClient CreateDisposableClient(Action<ClusterBuilder
         }
 
         public static DisposableMongoClient CreateDisposableClient(
-            Action<MongoClientSettings> clientSettingsConfigurator, 
+            Action<MongoClientSettings> clientSettingsConfigurator,
             bool useMultipleShardRouters = false)
         {
+            if (CoreTestConfiguration.Cluster.Description.Type != ClusterType.Sharded)
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.TestHelpers/DriverTestConfiguration.cs:117]: According to spec, this option makes sense only for sharded topologies

**Comment by rstam** [comment]
The timestamps on the commit are wrong. It "looks" like this PR has been pending for 6 days but it only just today showed up in my github,com/pulls queue.

**Comment by DmitryLukyanov** [comment]
> 
> 
> The timestamps on the commit are wrong. It "looks" like this PR has been pending for 6 days but it only just today showed up in my github,com/pulls queue.

yep, I created a draft PR to track evergreen patches which I've created for this ticket. Then I converted this PR to the usual PR

**Comment by rstam** [comment]
> yep, I created a draft PR to track evergreen patches which I've created for this ticket. Then I converted this PR to the usual PR

But the commit itself is dated 6 days ago, and I'm pretty sure you made changes this morning.

**Comment by DmitryLukyanov** [comment]
> 
> 
> > yep, I created a draft PR to track evergreen patches which I've created for this ticket. Then I converted this PR to the usual PR
> 
> But the commit itself is dated 6 days ago, and I'm pretty sure you made changes this morning.

yep, squashing issue :)

**Review by rstam — APPROVED** [review]
LGTM

**Review by rstam — CHANGES_REQUESTED** [review]
I'm reverting my LGTM because it is no longer clear to me that the original workaround or the changes in this pull request have anything to do with SERVER-41532.

**Comment by rstam** [comment]
We have determined that the workaround mentioned in CSHARP-2629 was actually a workaround for SERVER-40685 and not SERVER-41532.

The JIRA ticket has been updated accordingly.

**Review by rstam — APPROVED** [review]
Now that we have corrected the JIRA ticket to indicate that the workaround we are removing was actually for SERVER-40685 (which is fixed) and not SERVER-41532 (which is not fixed) I can LGTM this pull request again.

File: tests/MongoDB.Driver.TestHelpers/DriverTestConfiguration.cs [hunk 2]
```diff
@@ -113,9 +111,15 @@ public static DisposableMongoClient CreateDisposableClient(Action<ClusterBuilder
         }
 
         public static DisposableMongoClient CreateDisposableClient(
-            Action<MongoClientSettings> clientSettingsConfigurator, 
+            Action<MongoClientSettings> clientSettingsConfigurator,
             bool useMultipleShardRouters = false)
```
Comment (vincentkam) [inline tests/MongoDB.Driver.TestHelpers/DriverTestConfiguration.cs:115]: Just noting that most of our test runners call this function with the default value of false for `useMultipleShardRouters`. If I recall correctly, when we discussed this some time ago, we were okay with only testing against multiple shard routers only when the tests specifically called for them. (Confirmed with @jyemin offline).

**Review by vincentkam — APPROVED** [review]
Glad to see the workaround go! LGTM! 😸

**Pull request closed without merging**